libr/anal/fcn.c

```diff
@@ -275,19 +275,7 @@ void extract_arg (RAnal *anal, RAnalFunction *fcn, RAnalOp *op, const char *reg,
 	int len, lenmax=50;
 	char *sig = malloc (sizeof (char)*lenmax+1);
 	if(!sig) return;
-	strcpy(sig,",");
-	strncat (sig, reg, lenmax-1);
-	len = strlen(sig);
-	if (len > lenmax - 3) { /*<=============-i
-		1 for first colon		||
-		1 for the next colon		||
-		at least one for the sign	/|
-		total of 3 =====================_| */
-		return;
-	}
-	strncat (sig, ",", lenmax - len++);
-	strncat (sig, sign, lenmax - len);
-	//strcat (sig,",[4],");
+	snprintf (sig, lenmax, ",%s,%s", reg,sign);
```

Comment:
you can also use r_str_newf() or sdb_fmt() to avoid having to compute the buffer size by hand